integration_tests/src/main/python/date_time_test.py

```diff
@@ -467,13 +467,32 @@ def test_gettimestamp_ansi_exception():
 
 supported_date_formats = ['yyyy-MM-dd', 'yyyy-MM', 'yyyy/MM/dd', 'yyyy/MM', 'dd/MM/yyyy',
                           'MM-dd', 'MM/dd', 'dd-MM', 'dd/MM']
-@pytest.mark.parametrize('date_format', supported_date_formats, ids=idfn)
+@pytest.mark.parametrize('date_format', [date_gen], ids=idfn)
 @pytest.mark.parametrize('data_gen', date_n_time_gens, ids=idfn)
 @allow_non_gpu(*non_utc_allow)
-def test_date_format(data_gen, date_format):
+def test_date_format_for_date(data_gen, date_format):
     assert_gpu_and_cpu_are_equal_collect(
         lambda spark : unary_op_df(spark, data_gen).selectExpr("date_format(a, '{}')".format(date_format)))
 
+@pytest.mark.parametrize('date_format', supported_date_formats, ids=idfn)
+@pytest.mark.parametrize('data_gen', [timestamp_gen], ids=idfn)
+@pytest.mark.xfail(is_dst_time_zone(), reason="only support non-DST time zone, refer to https://github.com/NVIDIA/spark-rapids/issues/6839")
```

Comment:
Can we avoid ```xfail``` here? 

If  ```is_dst_time_zone = false```, test result equals without fallback.   
If  ```is_dst_time_zone = true```, test result equals with CPU operator fallback. This one can be combined with  ```test_date_format_for_date```.

Reply:
Using xfail is to record the xfail/xpass cases.
I'm OK to use skipif, but we may forget it in the future.

Reply:
can we use fallback check instead? I am nervous on ```xfail``` which hides errors.

Reply:
OK, will do it

Reply:
Done